### DISCUSSION TITLE
Battery Power View 🔋 🪫 ?

### DISCUSSION THREAD

**Discussion by hainezee** [SynthstromAudible/DelugeFirmware#2991] [Ideas]
Wondering if the amazing community dev team are thinking of improving the oled display info to show actually useful info?? I really don’t need to know I’m in song view/arranger view, but would love to know approx power remaining in my battery (the led indicator is pretty rubbish right??)… perhaps a percentage remaining display might be possible???

**Comment by duncanrmi** [comment]
yeppers. never had a power issue myself, but I'd be first in line for this.
also wondered- maybe this should be a new topic- if anyone's investigated putting a higher-voltage battery in the deluge so that it can power peripherals while on battery power.  I guess this would need a change to the charging arrangements, but physically at least it needn't be difficult...

**Comment by nikodemus** [comment]
@jamiefaye do you know if we have any kind of voltage metering even available? I suspect the LED is directly connected to the battery controller, and the CPU doesn't have any access - but might be wrong.

**Comment by m0nkmaster** [comment]
There is [code](https://github.com/SynthstromAudible/DelugeFirmware/blob/community/src/deluge/deluge.cpp#L155-L206) checking the voltage already, it is used for the colour of the led next to the power socket. I'm sure this could be extended to also add a power icon on the OLED screen.


```
int32_t numericReading = ADC.ADDRF;
// Apply LPF
int32_t voltageReading = numericReading * 3300;
int32_t distanceToGo = voltageReading - voltageReadingLastTime;
voltageReadingLastTime += distanceToGo >> 4;

// We only >> by 15 so that we intentionally double the value, because the incoming voltage is halved by a
// resistive divider already
batteryMV = (voltageReadingLastTime) >> 15;
```

**Comment by m0nkmaster** [comment]
<img width="477" alt="Screenshot 2025-01-22 at 23 46 44" src="https://github.com/user-attachments/assets/4694e0ee-478a-4e1e-a75e-cded63ab51ff" />

The data is there but relying on the voltage isn't ideal, the reading can fluctuate quite a lot.

**Reply by hainezee** [reply]
But better than not having?? What's your thoughts m0nkmaster?

**Reply by m0nkmaster** [reply]
I'm finding it quite useful, I did miss it when I put 1.2 FW back on. I'll dabble a bit further to see if I can make it more accurate.

**Reply by m0nkmaster** [reply]
@WIP https://github.com/SynthstromAudible/DelugeFirmware/pull/3310

**Comment by m0nkmaster** [comment]
Popup version demo attached.

https://github.com/user-attachments/assets/d3d23ea8-75a4-4ccc-bb34-e6f91636f031
